### ISSUE TITLE
Nest Router with Middleware not work

### ISSUE THREAD

**Issue by WClouds** [nestjsx/nest-router#5]
If I use nest-router, when I add a logger middleware for ApplicationModule, the middleware not work.
But when I remove nest-router, middleware worked. Is there any conflicts between nest-router and middleware?

```
// app.module.ts
import { Module, NestModule, MiddlewaresConsumer, RequestMethod } from '@nestjs/common';
import { RouterModule } from 'nest-router';
import { routes } from './routes';
import { SoftwareModule } from './software/software.module';
import { loggerMiddleware } from './logger.middleware';

@Module({
  imports: [RouterModule.forRoutes(routes), SoftwareModule],
})
export class ApplicationModule implements NestModule {
  configure(consumer: MiddlewaresConsumer): void {
    consumer
      .apply(loggerMiddleware)
      .forRoutes({
        path: '/api/v2',
        method: RequestMethod.ALL
      })
  }
}
```

```
// logger.middleware.ts
export const loggerMiddleware = (req, res, next) => {
  console.log(`Request...`);
  next();
};
```

```
// routes.ts
import { Routes } from 'nest-router';
import { SoftwareModule } from './software/software.module';

const v2_modules = [
  SoftwareModule,
]

const route = v2_modules.map(module => {
  return {
    path: '/api/v2',
    module,
  }
})

export const routes: Routes = route;
```

**Comment by shekohex** [comment]
Hi @WClouds 
Could you share a Minimal reproduction of the problem, so I can solve it.
but I just tested the simple `loggerMiddleware` with the example provided with Nest Router Repo, and it just works fine.

**Comment by Metheny80** [comment]
@shekohex I still can't get it to work.
The example in the repo sets up the middleware for the other modules in the main App Module, but in my case I have a few modules, and each sets up its own middleware.

For example, in the main App Module I set up the routes:

// app-module.ts

```
const routes: Routes = [
  {
    path: '/my',
    module: MyModule,
  },
];

@Module({
  imports: [
    RouterModule.forRoutes(routes)
 ])
export class AppModule { }
```

However when I do the following in MyModule, the middlewares are not actually being set up (called) for the module's routes:

// my-module.ts
`consumer.apply(MyMiddleware).forRoutes(MyController);`

// my-controller.ts
```
@Controller('path')
export class MyController {
  @Get()
  async action(): Promise<any> {
    return;
 }
```
Calling GET my/path doesn't trigger MyMiddleware.

I tried moving the routes definition to MyModule, but that doesn't work either.
When setting up the middleware like above, should the consumer take into account the prefix set in the AppModule?
Or any other idea why this doesn't work?
